test/function/config/wel_supported.cpp

```diff
@@ -335,6 +335,47 @@ namespace {
     EXPECT_ERROR_ID_EQ( STUMPLESS_ARGUMENT_EMPTY );
   }
 
+  TEST_F( WelSupportedTest, InvalidUtf8InsertionString ) {
+    const struct stumpless_entry *entry_result;
+    LPCSTR invalid_utf8_string = "\xc3\x28 invalid";
+
+    entry_result = stumpless_set_wel_insertion_string( insertion_entry, 0, invalid_utf8_string );
+    EXPECT_NULL( entry_result );
+  }
+
+  TEST_F( WelSupportedTest, InsertionStringNotModifiedWhenInvalidUtf8StringProvided ) {
+    LPCSTR valid_string = "valid string";
+    LPCSTR invalid_utf8_string = "\xc3\x28 invalid";
+    const struct stumpless_entry *result;
+
+    stumpless_set_wel_insertion_string( insertion_entry, 0, valid_string );
+    EXPECT_NO_ERROR;
+
+    result = stumpless_set_wel_insertion_string( insertion_entry, 0, invalid_utf8_string );
+    EXPECT_NULL( result );
+
+    LPCSTR actual_string = stumpless_get_wel_insertion_string( insertion_entry, 0 );
+    
+    ASSERT_STREQ( actual_string, valid_string );
+  }
+
+  /* 
+    Add the test case for invalid utf8 insertion string when insertion string is out of range
+  */
+
+  TEST_F ( WelSupportedTest, InsertionIndexNotModifiedWhenInRangeButNotAssigned ) {
+    const struct stumpless_entry *result;
+    LPCSTR invalid_utf8_string = "\xc3\x28 invalid";
+    LPCSTR valid_string = "valid string";
+
+    stumpless_set_wel_insertion_strings(insertion_entry, 2, invalid_utf8_string, NULL);
```

Comment:
add result and error id checks after this call

Reply:
Sure, I'm fixing the same. Can you elaborate on the insertion string being out of range part (from the issue #273) so that I can add this test to the commit as well, I could not get how a valid insertion string be out of range and the invalid one being in range.

Reply:
The language in the issue is a little confusing, sorry about that. You can achieve the state I was trying to describe by setting a valid insertion string at a high index (I would use index 3 or 4, since the fixture entry already has 2 insertion strings), and then setting the invalid one at index 2. Index 2 won't have been assigned yet, but will be in the already allocated range because of the higher string successfully set. Your test will need to make sure that it is still empty after the attempt to set it to an invalid string.

Reply:
Got it.